src/components/CopyAddressButton/index.tsx

```diff
@@ -0,0 +1,58 @@
+
+import * as React from 'react';
+import clsx from 'clsx';
+import {
+  DocumentDuplicateIcon,
+  CheckIcon
+} from '@heroicons/react/outline';
+
+import InterlayButtonBase, { Props as InterlayButtonBaseProps } from 'components/UI/InterlayButtonBase';
+import { copyToClipboard } from 'common/utils/utils';
+
+interface CustomProps {
+  address: string;
+}
+
+const COPY_ADDRESS_ICON_CLASSES = clsx(
+  'w-6',
+  'h-6'
+);
+
+const CopyAddressButton = ({
+  address,
+  className
+}: CustomProps & InterlayButtonBaseProps): JSX.Element => {
+  const [addressCopied, setAddressCopied] = React.useState<boolean>(false);
```

Comment:
@tomjeatt 
I think can create a reusable hook following https://twitter.com/benmvp/status/1374499450710757391.

Reply:
Yes, for now it's the only use case. But in blockchain products (specifically DeFi), this feature is really needed in many places so I expect we will need this for sure.

I used https://twitter.com/benmvp/status/1374499450710757391 before and it considers DX.